### ISSUE TITLE
Support for cylinder, capsule or urdf obstacles

### ISSUE THREAD

**Issue by newuhe** [tud-amr/fabrics#127]
Does fabrics support cylinder, capsule or urdf obstacles? It seems now only support static sphere or moving sphere and static cuboid obstacles. Also, if num of obstacles increases to 5 or more, the planning time is quite slow.

**Comment by newuhe** [comment]
If there is obstacle in the scene, the closer the arm is to the target pose, the slower it moves. If there is no obstacle, the arm will reach target pose quickly.

**Comment by maxspahn** [comment]
@newuhe You seem to have spotted the example with the point_robot, where we use the euclidean signed distance field, right? Currently, there is no example with a robotic arm and collision avoidance with an arm.

Fabrics are designed to move slow in the proximity of obstacles, that is considered safer. So your described behavior is expected. However, you can always tune up the attractor weight to increase the general speed. Also, the arm will always slow down approaching the target to avoid overshooting.

**Comment by newuhe** [comment]
I'm confused about definition of the distance between mesh and robot. Should it be signed distance filed? If so the collision avoidance is limited to current point view without consideration of 3D obstacle.

**Comment by maxspahn** [comment]
Indeed, it is taken the closest point of the robot to the obstacle and its gradient. This is why it is considered a reactive method with no time horizon. However. You can and should take multiple links of your robot for collision avoidance. For the planning time, can you give me an example of that?
